### PR TITLE
Fix the bug of feature filter evaluation when "requirement_type" is All

### PR DESCRIPTION
**Pull request by zhiyuanliang-ms**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/microsoft/FeatureManagement-Python/blob/main/CONTRIBUTING.md).**
- [x] Pull request includes test coverage for the included changes.

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.

### CODE REVIEW

File: featuremanagement/_featuremanager.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def is_enabled(self, feature_flag_id, **kwargs):
             filter_name = feature_filter[FEATURE_FILTER_NAME]
             if filter_name in self._filters:
                 if feature_conditions.requirement_type == REQUIREMENT_TYPE_ALL:
-                    if not self._filters[filter_name].evaluate(feature_filter[FEATURE_FILTER_PARAMETERS], **kwargs):
+                    if not self._filters[filter_name].evaluate(feature_filter, **kwargs):
```
Comment (mrm9084) [inline featuremanagement/_featuremanager.py:102]: @zhiyuanliang-ms, thanks for catching this. Can you update the async version too, so I don't have to do it in a different PR?

**Review by mrm9084 — CHANGES_REQUESTED** [review]

File: featuremanagement/_featuremanager.py (hunk 1 above)
Comment (zhiyuanliang-ms) [reply featuremanagement/_featuremanager.py:102]: @mrm9084 No problem.

**Review by mrm9084 — APPROVED** [review]
